numpy/random/mtrand/mtrand.pyx

```diff
@@ -1084,6 +1084,12 @@ cdef class RandomState:
 
         if p is not None:
             d = len(p)
+
+            atol = np.sqrt(np.finfo(np.float64).eps)
+            if isinstance(p, np.ndarray):
+                if np.issubdtype(p.dtype, np.floating):
+                    atol = max(atol, np.sqrt(np.finfo(p.dtype).eps))
+
             p = <ndarray>PyArray_ContiguousFromObject(p, NPY_DOUBLE, 1, 1)
```

Comment:
Looks like we could just import `array` and use that. We would still need to check casting.